apps/chat/src/components/Marketplace/TabRenderer.tsx

```diff
@@ -72,6 +80,39 @@ export const TabRenderer = ({ entities, isMobile }: TabRendererProps) => {
   }>();
   const [detailsModel, setDetailsModel] = useState<DialAIEntityModel>();
 
+  const displayedEntities = useMemo(() => {
+    const filteredEntities = allModels.filter(
+      (entity) =>
+        (doesEntityContainSearchTerm(entity, searchTerm) ||
+          (entity.version &&
+            doesEntityContainSearchTerm(
+              { name: entity.version },
+              searchTerm,
+            ))) &&
+        (selectedFilters[FilterTypes.ENTITY_TYPE].length
+          ? selectedFilters[FilterTypes.ENTITY_TYPE].includes(entity.type)
+          : true),
+    );
+
+    const entitiesForTab =
+      selectedTab === MarketplaceTabs.MY_APPLICATIONS
+        ? filteredEntities.filter((entity) =>
+            installedModels.some(({ id }) => id === entity.id),
```

Comment:
save installedModelsIds into constant or even better into `Set` (has)
```suggestion
            installedModelsIds.includes(entity.id),
```